src/wallet/wallet2.cpp

```diff
@@ -9336,17 +9336,22 @@ std::vector<size_t> wallet2::pick_preferred_rct_inputs(uint64_t needed_money, ui
 
   LOG_PRINT_L2("pick_preferred_rct_inputs: needed_money " << print_money(needed_money));
 
+  // pre-cache a "is this output to be considered" array to speed up tests later
+  std::vector<bool> spendable(m_transfers.size(), false);
+  for (size_t i = 0; i < m_transfers.size(); ++i)
+  {
+    const transfer_details& td = m_transfers[i];
+    if (!is_spent(td, false) && !td.m_frozen && td.is_rct() && is_transfer_unlocked(td) && td.m_subaddr_index.major == subaddr_account && subaddr_indices.count(td.m_subaddr_index.minor) == 1 && !td.m_key_image_partial)
+      if (td.amount() <= m_ignore_outputs_above && td.amount() >= m_ignore_outputs_below)
+        spendable[i] = true;
+  }
+
   // try to find a rct input of enough size
   for (size_t i = 0; i < m_transfers.size(); ++i)
   {
     const transfer_details& td = m_transfers[i];
-    if (!is_spent(td, false) && !td.m_frozen && td.is_rct() && td.amount() >= needed_money && is_transfer_unlocked(td) && td.m_subaddr_index.major == subaddr_account && subaddr_indices.count(td.m_subaddr_index.minor) == 1)
```

Comment:
The old and new conditions for this `if` branch are not the same. The old branch considered the transfer even with a partial key image, but this new branch does not (`... && !td.m_key_image_partial`). Is this intentional?

Reply:
Yes.